htlcswitch/hop/iterator.go

```diff
@@ -181,8 +186,26 @@ func MakeBlindingKit(processor BlindingProcessor,
 				return nil, err
 			}
 
+			// If we have an override for the blinding point for
+			// the next node, we'll just use it without tweaking
+			// (the sender intended to switch out directly for
+			// this blinding point). Otherwise, we'll tweak our
+			// blinding point to get the next ephemeral key.
+			nextEph, err := processor.NextEphemeral(blinding)
+			if err != nil {
+				return nil, err
+			}
+			routeData.NextBlindingOverride.WhenSome(
```

Comment:
Seems we should reverse the order of processing here? No need to actually do the computation above if we have an override. 

We can re-write this as: 
```
nextEph, err := routeData.NextBlindingOverride.UnwrapOrFuncErr(...)
```